File: src/main/java/com/GoldenMine/parser/parser/DefaultParser.java
package com.GoldenMine.parser.parser;

import com.GoldenMine.parser.Code;
import com.GoldenMine.parser.ParseContext;
import com.GoldenMine.parser.postposition.IPostPosition;
import com.GoldenMine.parser.postposition.JosaCommunity;
import com.GoldenMine.parser.postposition.JosaStorage;
import java.util.List;

public class DefaultParser implements IParser {
    private IParser childParser;

    public DefaultParser(IParser childParser) {
        this.childParser = childParser;
    }



    @Override
    public void parse(Code code, ParseContext context, int index) {
        if (childParser != null)
            childParser.parse(code, context, index);
        if(!context.isNoParse()) {
            String source = context.getSource();
            List<? extends IPostPosition> 격조사리스트 = JosaStorage.INSTANCE.getJosaList(JosaCommunity.격조사);

            context.getParsedMap().putAll(defaultParse(source, 격조사리스트, false, null));

        }
    }

//    private void addToContext(ParseContext context, IPostPosition postposition, String result) {
//        HashMap<String, List<Context>> map = context.getParsedMap();
//
//        List<Context> contextList;
//        if(map.containsKey(postposition.getType())) {
//            contextList = map.get(postposition.getType());
//        } else {
//            contextList = new ArrayList<>();
//            map.put(postposition.getType(), contextList);
//        }
//
//        contextList.add(new Context(result));
//    }
}

File: src/main/java/kr/goldenmine/parser/parser/DefaultParser.kt
package kr.goldenmine.parser.parser

import kr.goldenmine.parser.Code
import kr.goldenmine.parser.ParseContext
import kr.goldenmine.parser.postposition.JosaCommunity
import kr.goldenmine.parser.postposition.JosaStorage
import java.util.*

class DefaultParser(private val childParser: IParser?) : IParser {
    override fun parse(code: Code, parseContext: ParseContext, index: Int, debug: Boolean, metadata: List<Any>?) {
        childParser?.parse(code, parseContext, index, debug, metadata)
        if (!parseContext.isNoParse) {
            val source = parseContext.source
            val 격조사List = JosaStorage.getGyukJosaListWithBojo()

            parseContext.parsedMap.putAll(defaultParse(source, 격조사List, false, null))
            if(debug)
                println("DefaultParser: ${parseContext.parsedMap}")
        }
    }

    //    private void addToContext(ParseContext context, IPostPosition postposition, String result) {
    //        HashMap<String, List<Context>> map = context.getParsedMap();
    //
    //        List<Context> contextList;
    //        if(map.containsKey(postposition.getType())) {
    //            contextList = map.get(postposition.getType());
    //        } else {
    //            contextList = new ArrayList<>();
    //            map.put(postposition.getType(), contextList);
    //        }
    //
    //        contextList.add(new Context(result));
    //    }
}
